sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/policy/AzureMonitorRedirectCustomPolicy.java

```diff
@@ -0,0 +1,92 @@
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.net.HttpURLConnection;
+
+/**
+ * A HttpPipeline policy that retries when a HTTP Redirect is received as response.
+ */
+public class AzureMonitorRedirectCustomPolicy implements HttpPipelinePolicy {
+
+    private final ClientLogger logger = new ClientLogger(AzureMonitorRedirectCustomPolicy.class);
+    private final int maxRedirectRetries;
+    private String redirectedEndpointUrl;
+
+    /**
+     * Creates {@link AzureMonitorRedirectCustomPolicy} with default maxRedirectRetries values as 10.
+     */
+    public AzureMonitorRedirectCustomPolicy() {
+        maxRedirectRetries = 10;
+    }
+
+    /**
+     * Creates {@link AzureMonitorRedirectCustomPolicy} with provided maxRedirectRetries.
+     *
+     * @param maxRedirectRetries The maximum retries through pipeline when a redirect is received as response
+     */
+    public AzureMonitorRedirectCustomPolicy(int maxRedirectRetries) {
+        this.maxRedirectRetries = maxRedirectRetries;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return attemptRetry(context, next, context.getHttpRequest(), 0);
+    }
+
+    /**
+     *  Function to process through the HTTP Response received in the pipeline
+     *  and retry sending the request with new redirect url.
+     */
+    private Mono<HttpResponse> attemptRetry(final HttpPipelineCallContext context,
+                                            final HttpPipelineNextPolicy next,
+                                            final HttpRequest originalHttpRequest,
+                                            final int retryCount) {
+        // make sure the context is not modified during retry, except for the URL
+        context.setHttpRequest(originalHttpRequest.copy());
+        if (this.redirectedEndpointUrl != null) {
+            context.getHttpRequest().setUrl(this.redirectedEndpointUrl);
+        }
+        return next.clone().process()
+            .flatMap(httpResponse -> {
+                if (shouldRetryWithRedirect(httpResponse.getStatusCode(), retryCount)) {
+                    String responseLocation = httpResponse.getHeaderValue("Location");
+                    if (responseLocation != null) {
+                        this.redirectedEndpointUrl = responseLocation;
+                        return attemptRetry(context, next, originalHttpRequest, retryCount + 1);
+                    }
+                }
+                return Mono.just(httpResponse);
+            }).onErrorResume(err -> Mono.error(new RuntimeException(
+                String.format("Retry with redirect error details: %s", err.getMessage()),
+                err)));
```

Comment:
This error may not necessarily be because of redirects. It could be due to timeouts, bad requests etc. Also, there's no need to have `onErrorResume` here.